File: Entities/Material.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MaterialCtrl.Entities {
    public class Material
    {
        public int Id { get; set; }
        [Required, MaxLength(250)]
        public string PartNumber { get; set; }
        [Required, MaxLength(250)]
        public string PartName { get; set; }
        public int Revision { get; set; }
        public string Description { get; set; }
        public Category Category { get; set; }
        public UnitType Unit { get; set; }
        public Manufacturer Manufacturer { get; set; }
        public string ManufacturerItemNumber { get; set; }
        public ProcurementType ProcurementType { get; set; }
        public string Notes { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }
        public User CreatedBy { get; set; }
        public User ModifiedBy { get; set; }
    }
}

File: ClientApp/app/shared/material.ts
export class Material {
    // use http://www.json2ts.com/ to convert entities to TS interface
    id: number;
    partNumber: string;
    partName: string;
    revision: number;
    description?: any;
    category: any;
    unit?: any;
    manufacturer: any;
    manufacturerPartNumber: string;
    procurementType: any;
    notes: string;
    createdOn: Date;
    modifiedOn: Date;
    createdBy?: any;
    modifiedBy?: any;
}